graylog2-web-interface/src/components/lookup-tables/CacheTableEntry.test.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+import { render, screen } from 'wrappedTestingLibrary';
+import { BrowserRouter as Router } from 'react-router-dom';
+
+import { asMock } from 'helpers/mocking';
+
+import { CACHE, mockedUseScopePermissions } from './fixtures';
+import CacheTableEntry from './CacheTableEntry';
+
+jest.mock('hooks/useScopePermissions', () => (mockedUseScopePermissions));
+
+const renderedCTE = (scope: string) => {
+  const auxCache = { ...CACHE };
+  auxCache._scope = scope;
+
+  return render(
+    <Router><CacheTableEntry cache={auxCache} /></Router>,
+    {
+      container: document.body.appendChild(document.createElement('table')),
+    },
+  );
+};
+
+describe('CacheTableEntry', () => {
+  it('should show Loading spinner while loading scope permissions', async () => {
+    asMock(mockedUseScopePermissions).mockReturnValueOnce({
+      loadingScopePermissions: true,
+      scopePermissions: null,
+    });
+
+    renderedCTE('DEFAULT');
+
+    expect(screen.queryByRole('button', { name: /edit/i })).not.toBeInTheDocument();
+    expect(screen.getByText('Loading...')).toBeInTheDocument();
+  });
+
+  it('should show "edit" button', async () => {
```

Comment:
It looks like these calls do not have to be `async`.

Reply:
Alright, I am personally a big fan of using `await findByRole`, since it also fails if the element does not exist and it still works in case the component includes async code in the future. No need to change it for this PR, it is just a personal preference.